mantidproject/mantid
# Add free text box to error reporter

This PR adds a free text input box to the error reporter. The box allows users to input a message of up to 3200 characters, which will be stored in the database.

Note, this PR requires an edit of the existing database schema, to add a field for the `textBox`

**Report to:**  Anders

**To test:**

* Open MantidPlot
* Cause a crash by running the Segfault algorithm
* The error report dialog should pop up
* In the Name box enter Private and in the Email box enter private
* In the free text box write "Private text"
* Click the Don’t share any information box

-----

* Open MantidPlot
* Cause a crash by running the Segfault algorithm
* The error report dialog should pop up
* In the Name box enter Private and in the Email box enter private
* In the free text box write "Public text"
* Click the Share information

----

* Open MantidPlot
* Cause a crash by running the Segfault algorithm
* The error report dialog should pop up
* In the Name box enter Private and in the Email box enter private
* In the free text box write "Non-ID text"
* Click the Share Non-identifiable information

----

* Open an old version MantidPlot (3.12 or 3.13)
* Cause a crash by running the Segfault algorithm
* The error report dialog should pop up
* In the Name box enter Private and in the Email box enter private
* In the free text box write "Non-ID text"
* Click the Share Non-identifiable information

----


* Check with a database administrator that the correct values have been saved.
* Check that the information from old versions of Mantid is also saved.



Release notes: `docs/source/release/v3.14.0/error_reporter.rst`

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
